pkg/vm/vm_test.go

```diff
@@ -91,6 +91,61 @@ func TestSimpleRun(t *testing.T) {
 
 }
 
+func TestThrow(t *testing.T) {
+
+	// Program pushes 20 to the stack
+	// exits with an error
+
+	// Push(20)
+	//THROW
+
+	builder := stack.NewBuilder()
+	builder.EmitInt(20).EmitOpcode(stack.THROW)
+
+	// Pass program to VM
+	vm := NewVM(builder.Bytes())
+
+	// Runs vm with program
+	_, err := vm.Run()
+	assert.Equal(t, "the execution of the script program end with an error", err.Error())
```

Comment:
You should not compare errors using their strings; it's very fragile. For now, just check for an error. In a future PR, a custom error struct can implemented for the VM.

Reply:
:+1: